ingestion/src/metadata/ingestion/source/sql_source.py

```diff
@@ -102,6 +107,113 @@ def _get_table_description(schema: str, table: str, inspector: Inspector) -> str
+    try:
+        dbt_options = config.dbtConfig.gcsConfig
+        if config.dbtProvider.value == "gcs":
+            options = {
+                "credentials": {
+                    "type": dbt_options.type,
+                    "project_id": dbt_options.projectId,
+                    "private_key_id": dbt_options.privateKeyId,
+                    "private_key": dbt_options.privateKey,
+                    "client_email": dbt_options.clientEmail,
+                    "client_id": dbt_options.clientId,
+                    "auth_uri": dbt_options.authUri,
+                    "token_uri": dbt_options.tokenUri,
+                    "auth_provider_x509_cert_url": dbt_options.authProviderX509CertUrl,
+                    "client_x509_cert_url": dbt_options.clientX509CertUrl,
+                }
+            }
+        else:
+            options = {"credentials_path": dbt_options.__root__}
+        if store_gcs_credentials(options):
+            from google.cloud import storage
+
+            client = storage.Client()
+            for bucket in client.list_buckets():
+                for blob in client.list_blobs(bucket.name):
+                    if MANIFEST_FILE_NAME in blob.name:
+                        dbt_manifest = blob.download_as_string().decode()
+                    if CATALOG_FILE_NAME in blob.name:
+                        dbt_catalog = blob.download_as_string().decode()
+            return json.loads(dbt_catalog), json.loads(dbt_manifest)
+        else:
+            return None
+    except Exception as exc:
+        logger.debug(traceback.print_exc())
+        logger.debug(f"Error fetching dbt files {repr(exc)}")
+    return None
+
+
+def get_dbt_s3(config) -> Optional[Tuple[str, str]]:
+    try:
+        from metadata.utils.aws_client import AWSClient
+
+        aws_client = AWSClient(config.dbtConfig).get_resource("s3")
+        buckets = aws_client.buckets.all()
+        for bucket in buckets:
+            for bucket_object in bucket.objects.all():
+                if MANIFEST_FILE_NAME in bucket_object.key:
+                    dbt_manifest = bucket_object.get()["Body"].read().decode()
+                if CATALOG_FILE_NAME in bucket_object.key:
+                    dbt_catalog = bucket_object.get()["Body"].read().decode()
+        return json.loads(dbt_catalog), json.loads(dbt_manifest)
+    except Exception as exc:
+        logger.debug(traceback.print_exc())
+        logger.debug(f"Error fetching dbt files {repr(exc)}")
+    return None
+
+
+def get_dbt_details(config) -> Optional[Tuple[str, str]]:
+    try:
+        if config.dbtProvider:
+            if config.dbtProvider.value == "s3":
+                return get_dbt_s3(config)
+            elif "gcs" in config.dbtProvider.value:
+                return get_dbt_gcs(config)
+            elif config.dbtProvider.value == "http":
+                return get_dbt_http(config)
+            elif config.dbtProvider.value == "local":
+                return get_dbt_local(config)
```

Comment:
Let's raise an exception if no `if` statement gets matched, something like `Invalid DBT provider`. It should not happen based on the JSON schemas, but that should also help reading the code